### ISSUE TITLE
Exon count with multiple transcripts

### ISSUE THREAD

**Issue by kimrutherford** [pombase/website#1605]
In the query builder how should querying for exon count work if there are multiple transcripts?  Currently it only looks at the first transcript.

**Comment by ValWood** [comment]
I think we should only use the canonical transcript for exon count feature.
This query is mainly used to bin loci, so I don't think it makes sense to include loci twice.
We can document that behaviour.

Maybe we could have some way of identifying those where the exon number is *sometimes* different?

**Comment by kimrutherford** [comment]
> We can document that behaviour.

As a quick fix perhaps we could change the description to:
```
Retrieve genes by number of exons in the primary transcript:
```


https://github.com/pombase/pombase-config/blob/master/website/pombase_v2_config.json#L301

**Comment by ValWood** [comment]
Good idea!

**Comment by kimrutherford** [comment]
> As a quick fix perhaps we could change the description to:

OK, done.

**Comment by mah11** [comment]
OK, I'll update the documentation to match.

In the longer run, is there much chance that we'll see any demand for it to look at all transcripts, and return genes where any transcript matches the requested number of exons? (I bet there would be for bigger eukaryotes, where alternative transcripts can be a big deal.)

**Comment by ValWood** [comment]
I suppose if you absolutely wanted  transcript  with x exons, it might be useful to search all. However, I don't know if this is a query people would need often, if ever ( I only use it  very occasionally for binning by exon number, which is why I was keen to keep it locus specific). Users have used it to locate genes with many or few introns.

Because most UTR introns are not annotated the numbers are a bit wishy-washy anyway....

**Comment by kimrutherford** [comment]
Maybe we should change this text in the query builder again: "Retrieve genes by number of exons in the primary transcript"

Because currently it only counts coding exons.  It would be more precise to say:
```
Retrieve genes by number of coding exons in the primary transcript
```

**Comment by ValWood** [comment]
Right, I thought we only counted CDS exons, but then I found that RNAs were also included...

This is getting quite confusing :)

It might even be better to treat as 2 separate querries

Retrieve proteins by coding exon count
Retrieve non coding RNA by coding exon count

(not that I should be able to identify and annotate the exons in UTRs if the data is good enough)

**Comment by mah11** [comment]
> Retrieve non-coding RNA by coding exon count

No. We absolutely cannot use this label, because it makes no sense. A non-coding RNA, by definition, has no coding exons.

If someone is interested in any transcript that would get spliced, then all exons should count.

If they're interested in things with more than one coding exon, then UTR exons should not count ... and ncRNAs should also be excluded from results.

This could be presented as two queries or as one query type with an option to restrict to coding exons of protein-coding genes, or not. (I slightly prefer the latter.)

**Comment by ValWood** [comment]
For now, let's just add the query for proteins only.

Retrieve genes by number of coding exons in the primary transcript.
Then we can close this unless anyone asks for more. It is doubtful anyone will, it's a bit of an edge case query option.

**Comment by ValWood** [comment]
This seems to be just a text change?

**Comment by kimrutherford** [comment]
It needs this code change too:

> For now, let's just add the query for proteins only.

**Comment by kimrutherford** [comment]
>> This seems to be just a text change?
>
> It needs this code change too:
>
>> For now, let's just add the query for proteins only.

Note to self, we need to change to only count coding exons once we query for just for proteins:

> Retrieve genes by number of coding exons in the primary transcript.
